homeassistant/helpers/entity_registry.py

```diff
@@ -171,7 +171,9 @@ def _async_update_entity(self, entity_id, *, name=_UNDEF,
             changes['device_id'] = device_id
 
         if new_entity_id is not _UNDEF and new_entity_id != old.entity_id:
-            if self.async_is_registered(new_entity_id):
+            if (self.async_is_registered(new_entity_id) or new_entity_id in
```

Comment:
The entity registry has nothing to do with what is in `hass.states`. It looks also like you're trying to solve a symptom instead of a problem.